File: extensions/task-with-files/scripts/after-write-file.js
#!/usr/bin/env node
/**
 * @license
 * Copyright 2026 Google LLC
 * SPDX-License-Identifier: Apache-2.0
 */

/**
 * AfterWriteFile Hook
 * 
 * Reminds user to update progress.md after modifying files.
 */

const fs = require('fs');
const path = require('path');

const WORKSPACE_PATH = process.env.WORKSPACE_PATH || process.env.workspace_path || process.cwd();
const workingDirBase = path.join(WORKSPACE_PATH, '.agent_working_dir');
const currentTaskFile = path.join(workingDirBase, 'current_task.json');

/**
 * Get current task directory
 * @returns {string|null} - Task directory path or null
 */
function getCurrentTask() {
  if (!WORKSPACE_PATH) {
    return null;
  }

  if (!fs.existsSync(currentTaskFile)) {
    return null;
  }

  try {
    const data = JSON.parse(fs.readFileSync(currentTaskFile, 'utf-8'));
    return data.current || null;
  } catch (err) {
    return null;
  }
}

/**
 * Output JSON to stdout
 * @param {object} data - Data to output
 */
function outputJson(data) {
  console.log(JSON.stringify(data));
}

function main() {
  // Read stdin (required for hook protocol)
  let input_data = {};
  try {
    const stdin = fs.readFileSync(0, 'utf-8');
    input_data = JSON.parse(stdin);
  } catch (err) {
    outputJson({ decision: 'allow' });
    return;
  }

  const currentTask = getCurrentTask();

  if (!currentTask) {
    outputJson({ decision: 'allow' });
    return;
  }

  const toolInput = input_data.tool_input || {};
  const filePath = toolInput.file_path || '';

  let additionalContext;

  if (filePath) {
    try {
      const relPath = path.relative(WORKSPACE_PATH, filePath);
      additionalContext = `

---

**[task-with-files] File Modified: ${relPath}**

You just modified this file. Consider updating **progress.md** with what was accomplished.

Example format for progress.md:
\`\`\`
## ${relPath}

- [x] Description of change made
\`\`\`
**Current Task Folder:** \`${currentTask}\`
`;
    } catch (err) {
      additionalContext = `

---

**[task-with-files] You just modified a file.**

Consider updating **progress.md** with what was accomplished.
**Current Task Folder:** \`${currentTask}\`
`;
    }
  } else {
    additionalContext = `

---

**[task-with-files] You just modified a file.**

Consider updating **progress.md** with what was accomplished.
**Current Task Folder:** \`${currentTask}\`
`;
  }

  outputJson({
    decision: 'allow',
    hookSpecificOutput: { additionalContext },
  });
}

main();

File: extensions/task-with-files/scripts/after_write_file.py
#!/usr/bin/env python3
import json
import os
import sys
from pathlib import Path


def get_current_task(workspace_path: str):
    if not workspace_path:
        return None

    current_task_file = (
        Path(workspace_path) / ".agent_working_dir" / "current_task.json"
    )

    if not current_task_file.exists():
        return None

    try:
        with open(current_task_file, "r") as f:
            data = json.load(f)
        return data.get("current")
    except:
        return None


def main():
    workspace_path = os.environ.get("WORKSPACE_PATH", "")

    try:
        input_data = json.load(sys.stdin)
    except json.JSONDecodeError:
        print(json.dumps({"decision": "allow"}), flush=True)
        return

    current_task = get_current_task(workspace_path)

    if not current_task:
        print(json.dumps({"decision": "allow"}), flush=True)
        return

    # Ensure current_task is an absolute path
    current_task_path = current_task if Path(current_task).is_absolute() else str(Path(workspace_path) / ".agent_working_dir" / current_task)

    tool_input = input_data.get("tool_input", {})
    file_path = tool_input.get("file_path", "")

    if file_path:
        try:
            ws = Path(workspace_path)
            rel_path = Path(file_path).relative_to(ws)
        except ValueError:
            rel_path = Path(file_path)

        additional_context = f"""

---

**[task-with-files] File Modified: {rel_path}**

You just modified this file. Consider updating **progress.md** with what was accomplished.

Example format for progress.md:
```
## {rel_path}

- [x] Description of change made
```

**Current Task Folder:** `{current_task_path}`

> Temporary files (test files, scripts, scratch pads) created during this task can be written within the task folder to keep the workspace organized.
"""
    else:
        additional_context = f"""

---

**[task-with-files] You just modified a file.**

Consider updating **progress.md** with what was accomplished.

**Current Task Folder:** `{current_task_path}`

> Temporary files (test files, scripts, scratch pads) created during this task can be written within the task folder to keep the workspace organized.
"""

    print(
        json.dumps(
            {
                "decision": "allow",
                "hookSpecificOutput": {"additionalContext": additional_context},
            }
        ),
        flush=True,
    )


if __name__ == "__main__":
    main()
